datadotworld/client/api.py

```diff
@@ -441,6 +445,314 @@ def download_datapackage(self, dataset_key, dest_dir):
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SPARQL query
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sparql_get('username/test-dataset', query)
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sparql_api.sparql_get(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def sparql_post(self, dataset_key, query):
+        """Executes SPARQL queries against a dataset via POST
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SPARQL query
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sparql_post('username/test-dataset', query)
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sparql_api.sparql_post(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # Download Operations
+
+    def download_dataset(self, dataset_key):
+        """Return a .zip containing all files within the dataset as uploaded.
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.download_dataset('username/test-dataset')
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._download_api.download_dataset(owner_id, dataset_id)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def download_file(self, dataset_key, file):
+        """Return a .zip of a files within the dataset as uploaded.
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        file : str
+            File path to be returned
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.download_file('username/test-dataset', ['/my/local/example.csv'])
```

Comment:
Is the file param supposed to be a list? The docstring says `str`, but the example uses a list.